File: leetcode/java/26. Remove Duplicates from Sorted Array.java
import java.util.*;

class Solution {
    public int removeDuplicates(int[] nums) {
            TreeMap<Integer,Integer> map= new TreeMap<Integer,Integer>();
    for(int i=0;i<nums.length;i++){
      if(!map.containsValue(nums[i])){
        map.put(i,nums[i]);
      }
    }
    int i=0;
    for(Map.Entry m:map.entrySet()){
      nums[i]=(int)m.getValue();
      i++;
    }
    return map.size();
    }
}

File: leetcode/kotlin/26. Remove Duplicates from Sorted Array.kt
import java.util.*

internal class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        val map = TreeMap<Int, Int>()
        for (i in nums.indices) {
            if (!map.containsValue(nums[i])) {
                map[i] = nums[i]
            }
        }
        var i = 0
        for ((_, value) in map) {
            nums[i] = value
            i++
        }
        return map.size
    }
}